File: 1742-widest-vertical-area-between-two-points-containing-no-points/widest-vertical-area-between-two-points-containing-no-points.py
class Solution:
    def maxWidthOfVerticalArea(self, points: List[List[int]]) -> int:
        sort = sorted(points, key=lambda x: x[0])
        
        M=0
        for i in range(len(points)-1):
            val=sort[i+1][0]-sort[i][0]
            if val>M:
                M=val
        return M

File: 1742-widest-vertical-area-between-two-points-containing-no-points/widest-vertical-area-between-two-points-containing-no-points.cpp
class Solution {
public:
    int maxWidthOfVerticalArea(vector<vector<int>>& points) {
        std::sort(points.begin(), points.end(), [](const vector<int>&a, const vector<int>&b){
            return a[0]<b[0];
        });
    
    int cur_max=0;
    
    for (int i=0;i<points.size()-1;i++){
        int cur_val = points[i+1][0]-points[i][0];
        cur_max = std::max(cur_val, cur_max);
    }
    return cur_max;
    }

};